gamemodes/cinema/gamemode/modules/theater/sh_video.lua

```diff
@@ -206,15 +206,20 @@ if SERVER then
 end
 
 if CLIENT then
+    local TEMP_TRUST = TEMP_TRUST or {}
+    local CURRENTKEY = nil
+    
     function VIDEO:ShouldTrust()
-        return self:Service():ShouldTrust(self) or self.FORCETRUST
+        return self:Service():ShouldTrust(self) or (CURRENTKEY and TEMP_TRUST[CURRENTKEY]) or GetConVar("cinema_trust_videos") or self.FORCETRUST
     end
 
     --used by services
     function TRUSTED_VIDEO_HOST(k)
-        k = k:lower():gsub("https", "http"):gsub("http://", ""):gsub("www.", "")
+        local t = url.parse2(k)
+        if not t.host then return false end
+        CURRENTKEY = t.host
 
-        return k:StartWith("swampservers.net/") or k:StartWith("dropbox.com/") or k:StartWith("puu.sh/") or k:StartWith("discord.gg/") or k:StartWith("discord.com/")
+        return t.host == "www.swampservers.net" or t.host == "www.dropbox.com" or t.host == "www.puu.sh" or t.host == "cdn.discordapp.com"
```

Comment:
Wait so it puts www on everything, even if the url doesn't have www?

Reply:
the host member has the www in the string when I tested it so it's www unless there's a special case like cdn.discordapp.com